aws/resource_aws_elasticsearch_domain.go

```diff
@@ -137,6 +137,38 @@ func resourceAwsElasticSearchDomain() *schema.Resource {
 					},
 				},
 			},
+			"vpc_options": {
+				Type:     schema.TypeList,
+				Optional: true,
+				Elem: &schema.Resource{
+					Schema: map[string]*schema.Schema{
+						"security_group_ids": {
+							Type:     schema.TypeSet,
+							Optional: true,
+							Elem:     &schema.Schema{Type: schema.TypeString},
+							Set:      schema.HashString,
+						},
+						"subnet_ids": {
+							Type:     schema.TypeSet,
+							Optional: true,
+							Elem:     &schema.Schema{Type: schema.TypeString},
+							Set:      schema.HashString,
+						},
+					},
+				},
+			},
+			"vpc_availability_zones": {
+				Type:     schema.TypeSet,
+				Optional: true,
+				Computed: true,
+				Elem:     &schema.Schema{Type: schema.TypeString},
+				Set:      schema.HashString,
+			},
+			"vpc_id": {
```

Comment:
Is there any particular reason for keeping those two fields outside of `vpc_options` and drift away from the API?

Reply:
SubnetIds are globally unique. This behavior of submitting just subnets is seen elsewhere (ie [ALB](https://docs.aws.amazon.com/elasticloadbalancing/latest/APIReference/API_CreateLoadBalancer.html)) and should be familiar to anyone used to using VPC. 

Filtering by VPC is often just client-side to narrow down subnet list.

There are exceptions of course, for resources such as security groups that go into a VPC but not into a subnet.

Reply:
@handlerbot understood, but I don't think that gives us a reason to drift away from the API. They can both live nested under `vpc_options` if I understand correctly?

Additionally we should remove `Optional` from both mentioned fields as the user cannot specify them in the config.

Reply:
> I don't see a way to export schema sub-members as top-level attributes, other than defining them twice and exporting them twice?

I'm not sure I follow - the user can access nested attributes of any `TypeList` like this:

```
aws_elasticsearch_domain.vpc_options.0.vpc_id
```
the index may feel a bit weird, but that's how most other resources do it, so I'd just stick with it. It's something we plan to address in core/HCL eventually.